openrouteservice/src/main/java/org/heigit/ors/isochrones/builders/fast/FastIsochroneMapBuilder.java

```diff
@@ -0,0 +1,743 @@
+
+            // Use a default length (~1000m)
+            return 0.010;
+        }
+
+        double intervalDegrees = GeomUtility.metresToDegrees(maxRadius);
+        double maxLength = (intervalDegrees / 100f) * smoothingFactor;
+
+        if (maxLength < minimumDistance)
+            maxLength = minimumDistance;
+        return maxLength;
+    }
+
+    private void createPolyFromPoints(Set<Geometry> activeCellHulls, GeometryCollection points, double maxRadius, float smoothingFactor) {
+        if (points.isEmpty())
+            return;
+        LinearRing ring = null;
+        Geometry concaveHull;
+        try {
+            ConcaveHull ch = new ConcaveHull(points, convertSmoothingFactorToDistance(smoothingFactor, maxRadius), false);
+            concaveHull = ch.getConcaveHull();
+            if (concaveHull instanceof Polygon) {
+                ring = (LinearRing) ((Polygon) concaveHull).getExteriorRing();
+                List<Coordinate> coordinates = new ArrayList<>(ring.getNumPoints());
+                for (int i = 0; i < ring.getNumPoints(); i++) {
+                    coordinates.add(ring.getCoordinateN(i));
+                    if (i < ring.getNumPoints() - 1) {
+                        splitEdge(ring.getPointN(i).getY(),
+                                ring.getPointN(i + 1).getY(),
+                                ring.getPointN(i).getX(),
+                                ring.getPointN(i + 1).getX(),
+                                coordinates,
+                                minEdgeLengthLimit,
+                                maxEdgeLengthLimit);
+                    }
+                }
+                coordinates.add(ring.getCoordinateN(0));
+                concaveHull = geomFactory.createPolygon(coordinates.toArray(new Coordinate[0]));
+                activeCellHulls.add(concaveHull);
+            } else
+                activeCellHulls.add(concaveHull);
+        } catch (Exception e) {
+            if (isLogEnabled()) LOGGER.debug(e.getMessage());
+        }
+    }
+
+    private void addIsochrone(IsochroneMap isochroneMap, GeometryCollection points, double isoValue, double maxRadius, double meanRadius, float smoothingFactor) {
+        if (points.isEmpty())
+            return;
+        Polygon poly = null;
+        try {
+            ConcaveHull ch = new ConcaveHull(points, convertSmoothingFactorToDistance(smoothingFactor, maxRadius), false);
+            Geometry geom = ch.getConcaveHull();
+
+            if (geom instanceof GeometryCollection) {
+                GeometryCollection geomColl = (GeometryCollection) geom;
+                if (geomColl.isEmpty())
+                    return;
+            }
+
+            poly = (Polygon) geom;
+        } catch (Exception e) {
+            return;
+        }
+        isochroneMap.addIsochrone(new Isochrone(poly, isoValue, meanRadius));
+    }
+
+    public Boolean addPoint(List<Coordinate> points, Quadtree tree, double lon, double lat, boolean checkNeighbours) {
```

Comment:
It should, but it would require a rework of the ConcaveBallsIsochroneMapBuilder as well. Generally, it is possible to merge all the isochrone algorithms and make use of abstract classes for this. It was not scope of this project.